net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -506,44 +617,110 @@ mwan3_delete_iface_iptables()
 
 mwan3_create_iface_route()
 {
-	local id via metric V V_ IP
+	local id via metric V V_ IP family
+	local iface device cmd true_iface
 
-	config_get family "$1" family ipv4
-	mwan3_get_iface_id id "$1"
+	iface=$1
+	device=$2
+	config_get family "$iface" family ipv4 device
+	mwan3_get_iface_id id "$iface"
 
 	[ -n "$id" ] || return 0
 
+	mwan3_get_true_iface true_iface $iface
 	if [ "$family" = "ipv4" ]; then
-		V=4
 		V_=""
 		IP="$IP4"
 	elif [ "$family" = "ipv6" ]; then
-		V=6
 		V_=6
 		IP="$IP6"
-	else
-		return
 	fi
+	mwan3_get_true_iface true_iface $iface
 
-	if ubus call network.interface.${1}_${V} status &>/dev/null; then
-		network_get_gateway${V_} via "${1}_${V}"
-	else
-		network_get_gateway${V_} via "$1"
-	fi
+	network_get_gateway${V_} via "$true_iface"
 
 	( [ -z "$via" ] || [ "$via" = "0.0.0.0" ] || [ "$via" = "::" ] ) && unset via
 
-	network_get_metric metric "$1"
+	network_get_metric metric "$true_iface"
 
 	$IP route flush table "$id"
-	$IP route add table "$id" default \
+	cmd="$IP route add table $id default \
 	     ${via:+via} $via \
 	     ${metric:+metric} $metric \
-	     dev "$2"
-	mwan3_rtmon_ipv${V}
+	     dev $2"
+	$cmd || LOG warn "ip cmd failed $cmd"
+
+}
+
+mwan3_add_non_default_iface_route()
+{
+	local tid route_line family IP family id
+	config_get family "$1" family ipv4 device
+	mwan3_get_iface_id id "$1"
 
```

Comment:
dup `family` vars?
and typo `device`

Reply:
yes, thank you for the all comments. Just pushed fixes.